### PR TITLE
Add pop up and zoom in effect for polygon layer

### PR DESCRIPTION
**Pull request by XinyueHan7**

### CODE REVIEW

File: src/app/components/map/core-map/core-map.component.ts [hunk 1]
```diff
@@ -1,19 +1,20 @@
...
 @Component({
   selector: 'app-core-map',
   templateUrl: './core-map.component.html',
   styleUrls: ['./core-map.component.css']
 })
 export class CoreMapComponent implements OnInit {
-  constructor(private timeService: TimeService, private fireService: FireService, private tweetService: TweetService) {
-
+  constructor(private mapService: MapService, private timeService: TimeService, private fireService: FireService, private tweetService: TweetService) {
+    this.mapService.zoomIn.subscribe(this.zoomInToPolygon);
```
Comment (Yicong-Huang) [inline src/app/components/map/core-map/core-map.component.ts]: change the event emitter name to `zoomInToPolygonEvent`

Let's use `Event` suffix for event emitters

File: src/app/components/map/core-map/core-map.component.ts [hunk 2]
```diff
@@ -100,5 +101,7 @@ export class CoreMapComponent implements OnInit {
     this.map = map;
   }
 
-
+  zoomInToPolygon = ({zoomInBoundaries}) => {
```
Comment (Yicong-Huang) [inline src/app/components/map/core-map/core-map.component.ts]: why wrap the zoomInBoundaries in another object?

File: src/app/components/map/popup-box/popup-box.component.html [hunk 3]
```diff
@@ -1,5 +1,5 @@
-<div class="popup">Click me!
+<div class="popup">
 
-  <button (click)="click()" id="try">click</button>
+  <button (click)="click()" id="try">Zoom in</button>
   <span class="popuptext" id="myPopup">Popup text...</span>
```
Comment (Yicong-Huang) [inline src/app/components/map/popup-box/popup-box.component.html]: what is this for?

File: src/app/components/map/popup-box/popup-box.component.ts [hunk 4]
```diff
@@ -1,20 +1,34 @@
...
 })
 export class PopupBoxComponent implements OnInit {
-  constructor(private fireService: FireService) {
+  @ Input() fireId: string;
+
+  constructor(private fireService: FireService, private mapService: MapService) {
   }
 
   ngOnInit() {
+    this.fireId = this.fireId;
```
Comment (Yicong-Huang) [inline src/app/components/map/popup-box/popup-box.component.ts]: what is this line for?

File: src/app/services/fire/fire.service.ts [hunk 5]
```diff
@@ -10,11 +10,10 @@ import {environment} from '../../../environments/environment';
   providedIn: 'root'
 })
 export class FireService {
-
+  public fireBoundingData;
+  public fireIdClicked = new EventEmitter();
```
Comment (Yicong-Huang) [inline src/app/services/fire/fire.service.ts]: use `Event` for suffix

File: src/app/services/fire/fire.service.ts [hunk 6]
```diff
@@ -39,8 +38,16 @@ export class FireService {
...
     }));
   }
+
+  getFireBoundingBox(id): Observable<object> {
+    return this.searchFirePolygon(id, 2);
+  }
+
+  sendFireId(id) {
+    console.log('sending fire id');
+    this.fireIdClicked.next({fireId: id});
```
Comment (Yicong-Huang) [inline src/app/services/fire/fire.service.ts]: no need to wrap, why not just next(id)?

File: src/app/services/map/map.service.ts [hunk 7]
```diff
@@ -83,4 +85,8 @@ export class MapService {
     return this.http.get(`http://${environment.host}:${environment.port}/tweet/tweet-from-id`,
       {params: new HttpParams().set('tweet_id', id)});
   }
+
+  zoom(firePolygonLL) {
```
Comment (Yicong-Huang) [inline src/app/services/map/map.service.ts]: not sure what does function `zoom` do.  it should serve as a communication function right? not actually doing zoom

**Review by Yicong-Huang — CHANGES_REQUESTED** [review]
the call stack logic is correct. 

@Input can only be used for parent class with child classes.

File: src/app/services/map/map.service.ts (hunk 7 above)
Comment (XinyueHan7) [reply src/app/services/map/map.service.ts]: This zoom is not doing the zoomin, it serves as a communication function to inform the coremap to do the zoomin

File: src/app/components/map/popup-box/popup-box.component.html (hunk 3 above)
Comment (XinyueHan7) [reply src/app/components/map/popup-box/popup-box.component.html]: Will delete the id, the popup text is for adding text in the popup box later

**Review by Yicong-Huang — APPROVED** [review]
LGTM

**Pull request merged**